chore: add 0.3 release note (#426)
web-infra-dev/rspack-website: docs/en/blog/announcing-0.3.mdx

```js
const path = require('path');
const config = {
  module: {
    rules: [
      {
        test: /\.jsx$/,
        use: {
          loader: 'builtin:swc-loader',
          options: {
            // Enable source map
            sourceMap: true,
            jsc: {
              parser: {
                syntax: 'ecmascript',
                jsx: true,
              },
              transform: {
                react: {
                  pragma: 'React.createElement',
                  pragmaFrag: 'React.Fragment',
                  throwIfNamespace: true,
                  development: false,
                  useBuiltins: false,
                },
              },
            },
          },
        },
        type: 'javascript/auto',
      },
    ],
  },
};
module.exports = config;
```

> You can refer to builtin-swc-loader for more examples. Currently, built-in: swc-loader still has limitations, such as not supporting wasm plugins, etc. Rspack will continue to iterate and support more features of built-in: swc-loader in future versions.

Reviewer: Can we release this until it's fully supported? Since It does not currently support inline loader syntax now(This is a PR for this)

Author: it seems inline loader can be merged before 0.3 release？

Reviewer: There're still some tasks to be done. Not expected to be available on 0.3

Author: seems done now

  builtins: {
    postcss: {
      pxtorem: {
        rootValue: 50,
      },
    },
  },
};
```

• After:

```js
module.exports = {
  module: {
    rules: [
      {
        test: /\.css$/,
        use: [
          {
            loader: 'postcss-loader',
            options: {
              postcssOptions: {
                plugins: [
                  [
                    'postcss-plugin-px2rem',
                    {
                      rootValue: 100,
                    },
                  ],
                ],
              },
            },
          },
        ],
      },
    ],
  },
};
```

### Removal of Built-in CSS Autoprefixer Functionality

To align better with webpack’s CSS handling, Rspack removes the built-in autoprefixer functionality in 0.3. You can use `postcss-loader` to achieve `autoprefixer`.

```js
module.exports = {
  module: {
    rules: [
      {
        test: /\.css$/,
        use: [
          {
            loader: 'postcss-loader',
            options: {
              postcssOptions: {
                plugins: [['autoprefixer']],
              },
            },
          },
        ],
        type: 'css',
      },
    ],
  },
};
```

> You can refer to the postcss-loader-example for a complete example.

Reviewer 2: ```suggestion
You can refer to the [examples/postcss-loader](https://github.com/web-infra-dev/rspack/blob/main/examples/postcss-loader/rspack.config.js) for a complete example.
```

### Access to Internal Modules Restricted

Due to the current instability of the internal module API in Rspack, directly accessing the internal modules can easily lead to breaking changes. Therefore, Rspack restricts the ability to directly access internal modules and only supports accessing Rspack’s API from the root module.

• Before:

```js
import { Stats } from '@rspack/core/dist/stats'; // not supported since 0.3
```

• After:

```js
import { Stats } from '@rspack/core';
```

## Major Feature Updates

### Web Workers Support

Rspack natively supports Web Workers, which means you can use Web Workers out of the box without using worker-loader. Here is how to use it:

```js
new Worker(new URL('./worker.js', import.meta.url));
new Worker(new URL('./worker.js', import.meta.url), {
  name: 'my-worker',
});
```

> For more information about web workers support, see web workers.

Reviewer 2: ```suggestion
For more information about Web Workers support, see [Web Workers](guide/web-workers).
```

> ### builtin:swc-loader Support

Reviewer 2: ```suggestion
### `builtin:swc-loader` Support
```

> Although Rspack provides many swc compilation configuration options, these configurations are global and cannot fulfill the requirement of using different swc conversion logic for different modules. Therefore, Rspack supports `builtin:swc-loader` to provide more fine-grained swc conversion configuration. Compared to the JavaScript version of `swc-loader`, `builtin:swc-loader` has better performance. You can use `builtin:swc-loader` as follows:

Reviewer 2: ```suggestion
Although Rspack provides many SWC compilation configuration options, these configurations are global and cannot fulfill the requirement of using different swc transformation logic for different modules. Therefore, Rspack supports `builtin:swc-loader` to provide more fine-grained SWC transformation configuration. Compared to the JavaScript version of `swc-loader`, `builtin:swc-loader` has better performance. You can use `builtin:swc-loader` as follows:
```

```js
const path = require('path');
const config = {
  module: {
    rules: [
      {
        test: /\.jsx$/,
        use: {
          loader: 'builtin:swc-loader',
          options: {
            // Enable source map
            sourceMap: true,
            jsc: {
              parser: {
                syntax: 'ecmascript',
                jsx: true,
              },
              transform: {
                react: {
                  pragma: 'React.createElement',
                  pragmaFrag: 'React.Fragment',
                  throwIfNamespace: true,
                  development: false,
                  useBuiltins: false,
                },
              },
            },
          },
        },
        type: 'javascript/auto',
      },
    ],
  },
};
module.exports = config;
```

> You can refer to builtin-swc-loader for more examples. Currently, built-in: swc-loader still has limitations, such as not supporting wasm plugins, etc. Rspack will continue to iterate and support more features of built-in: swc-loader in future versions.

Reviewer: ```suggestion
You can refer to [builtin-swc-loader](https://github.com/web-infra-dev/rspack/tree/main/examples/builtin-swc-loader) for more examples. Currently, `builtin:swc-loader` still has limitations, such as not supporting wasm plugins, etc. Rspack will continue to iterate and support more features of `builtin:swc-loader` in future versions.
```

> You can refer to [builtin-swc-loader](https://github.com/web-infra-dev/rspack/tree/main/examples/builtin-swc-loader) for more examples. Currently, `builtin:swc-loader` still has limitations, such as not supporting wasm plugins, etc. Rspack will continue to iterate and support more features of `builtin:swc-loader` in future versions.

Reviewer 2: ```suggestion
You can refer to [examples/builtin-swc-loader](https://github.com/web-infra-dev/rspack/tree/main/examples/builtin-swc-loader) for more examples. Currently, `builtin:swc-loader` still has limitations, such as not supporting wasm plugins, etc. Rspack will continue to iterate and support more features of `builtin:swc-loader` in future versions.
```

### Improved Profile Support

Performance optimization is a common requirement in business support. To reduce the cost of performance optimization for businesses, we have improved the experience of Rspack Profile. You can generate profile-related files for performance optimization by using the RSPACK_PROFILE environment variable.

```sh
$ RSPACK_PROFILE=ALL rspack build
```

> For more detailed information about Profile, see Performance Profiling.

Reviewer 2: ```suggestion
For more detailed information about Profile, see [Performance Profiling](/guide/profile).
```

Alignment with More APIs

> • splitChunks.chunks supports regex.
> • Supports splitChunk.\{cacheGroup\}.type.
> • Supports splitChunk.\{cacheGroup\}.idHint.
> • Supports ensureChunkConditionsPlugin.
> • rule.use supports functions.
> • Supports configuration.profile.

Reviewer 2: ```suggestion
- `splitChunks.chunks` supports regex.
- Supports `splitChunk.\{cacheGroup\}.type`.
- Supports `splitChunk.\{cacheGroup\}.idHint`.
- Supports `ensureChunkConditionsPlugin`.
- `rule.use` supports functions.
- Supports `configuration.profile`.
```

### More Hook and Plugin Support

Compared to version 0.2, we have implemented more plugin APIs and made compatibility improvements for more plugins in version 0.3. At the same time, we have refined the plugin API support progress of webpack, making the support progress of plugin APIs transparent. You can track the implementation progress of plugin APIs here: [plugin-api-progress](https://github.com/orgs/web-infra-dev/projects/9).

### Alignment with Webpack Architecture

In version 0.3, we have further optimized the alignment with the webpack architecture, migrating from the original AST-based codegen architecture to the string transformation-based architecture. This alignment work further ensures that Rspack can align with more Hook APIs of Webpack during the codegen stage to be compatible with more community plugins.

### Rspack Ecosystem

> Starting from version 2.0, Rspack provides support for vue-loader. However, creating a complete Vue.js CLI solution based on vue-loader can be a complex task. To simplify the development of Vue.js applications using Rspack, we offer the [Modern.js Builder](), which provides an out-of-the-box Vue.js engineering solution called Modern.js Vue.js Support. This solution helps developers easily develop Vue.js applications using Rspack.

Reviewer 3: Missing link
Modern.js Builder: https://modernjs.dev/builder/en
Modern.js Vue.js Support: https://modernjs.dev/builder/en/guide/framework/vue3.html

Reviewer: ```suggestion
Starting from version 2.0, Rspack provides support for vue-loader. However, creating a complete Vue.js CLI solution based on vue-loader can be a complex task. To simplify the development of Vue.js applications using Rspack, we offer the [Modern.js Builder](https://modernjs.dev/builder/en), which provides an out-of-the-box Vue.js engineering solution called [Modern.js Vue.js Support](https://modernjs.dev/builder/en/guide/framework/vue3.html). This solution helps developers easily develop Vue.js applications using Rspack.
```
